mimic/rest/maas_api.py

```diff
@@ -1251,3 +1301,122 @@ def multiplot(self, request, tenant_id):
         request.setResponseCode(200)
         self._audit('rollups', request, tenant_id, status, content)
         return json.dumps({'metrics': metrics_replydata})
+
+
+@implementer(IAPIMock, IPlugin)
+@attributes(["maas_api"])
+class MaasControlApi(object):
+    """
+    This class registers the MaaS controller API in the service catalog.
+    """
+    def catalog_entries(self, tenant_id):
+        """
+        List catalog entries for the MaaS API.
+        """
+        return [
+            Entry(
+                tenant_id, "rax: monitor", "cloudMonitoringControl",
+                [
+                    Endpoint(tenant_id, region, text_type(uuid4()),
+                             "v1.0")
+                    for region in self.maas_api._regions
+                ]
+            )
+        ]
+
+    def resource_for_region(self, region, uri_prefix, session_store):
+        """
+        Get an :obj:`twisted.web.iweb.IResource` for the given URI prefix;
+        implement :obj:`IAPIMock`.
+        """
+        maas_controller = MaasController(api_mock=self,
+                                         session_store=session_store,
+                                         region=region)
+        return maas_controller.app.resource()
+
+
+@attributes(["api_mock", "session_store", "region"])
+class MaasController(object):
+    """
+    Klein routes for MaaS control API.
+    """
+
+    def _entity_cache_for_tenant(self, tenant_id):
+        """
+        Retrieve the M_cache object containing all objects created so far
+        """
+        return (self.session_store.session_for_tenant_id(tenant_id)
+                .data_for_api(self.api_mock.maas_api,
+                              lambda: collections.defaultdict(MCache))[self.region])
```

Comment:
Codecov gives this 1/2 because tests never execute inside the lambda. Any ideas how to work around this?

cc @lekhajee